platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/StsRealm.java

```diff
@@ -705,19 +777,8 @@ public void setSamlAssertionValidator(SamlAssertionValidator samlAssertionValida
 
     @Override
     public boolean doCredentialsMatch(AuthenticationToken token, AuthenticationInfo info) {
-      if (token instanceof SAMLAuthenticationToken) {
-        Object oldToken = token.getCredentials();
-        Object newToken = info.getCredentials();
-        return oldToken.equals(newToken);
-      } else if (token instanceof STSAuthenticationToken) {
-        String xmlCreds = ((STSAuthenticationToken) token).getCredentialsAsString();
-        if (xmlCreds != null && info.getCredentials() != null) {
-          return xmlCreds.equals(info.getCredentials());
-        }
-      } else {
-        if (token.getCredentials() != null && info.getCredentials() != null) {
-          return token.getCredentials().equals(info.getCredentials());
-        }
+      if (token.getCredentials() != null && info.getCredentials() != null) {
+        return token.getCredentials().equals(info.getCredentials());
```

Comment:
📘 Credentials can be directly compared since the credentials are (supposed to be) of type `PrincipalCollection`.